packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -617,11 +634,26 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
 
   private handleStateChange = (changes: StateChangeOptions<ShorthandValue>) => {
     if (changes.isOpen !== undefined && changes.isOpen !== this.state.open) {
-      this.trySetStateAndInvokeHandler('onOpenChange', null, { open: changes.isOpen })
+      const newState: { open: boolean; highlightedIndex?: number } = { open: changes.isOpen }
+
+      if (changes.isOpen) {
+        const highlightedIndexOnArrowKeyOpen = this.getHighlightedIndexOnArrowKeyOpen(changes)
+        if (_.isNumber(highlightedIndexOnArrowKeyOpen)) {
+          newState.highlightedIndex = highlightedIndexOnArrowKeyOpen
+        }
+        if (!this.props.search) {
+          this.listRef.current.focus()
+        }
+      } else {
+        newState.highlightedIndex = this.getHighlightedIndexOnClose()
+      }
+
+      this.trySetState(newState)
+      _.invoke(this.props, 'onOpenChange', null, { ...this.props, ...newState })
```

Comment:
logic on lines 651 and 652 was encapsulated in the method that used to be at line 620; can you please use that again?

```typescript
this.trySetStateAndInvokeHandler('onOpenChange', null, newState)
```

Reply:
thanks, will do